obspy/io/mseed/core.py

```diff
@@ -865,20 +865,19 @@ def record_handler(record, reclen, _stream):
         # Determine if a blockette 100 will be needed to represent the input
         # sample rate or if the sample rate in the fixed section of the data
         # header will suffice (see ms_genfactmult in libmseed/genutils.c)
-        #
-        # It is definitely necessary for very large or small sampling rates (
-        # using libmseed for the test would result in an overflow).
-        if trace.stats.sampling_rate >= 32727.0 or \
-                trace.stats.sampling_rate <= (1.0 / 32727.0):
+        use_blkt_100 = False
+
+        _factor = C.c_int16()
+        _multiplier = C.c_int16()
+        _retval = clibmseed.ms_genfactmult(
+            trace.stats.sampling_rate, C.pointer(_factor),
+            C.pointer(_multiplier))
+        # Use blockette 100 if ms_genfactmult() failed.
```

Comment:
that's good. It avoids the redefinition of the libmseed internal sampling rate limits!